### PR TITLE
Refactor: From JVM to Native using Quarkus

### PR DESCRIPTION
**Pull request by ingvaar**
> This is a complete project refactor.

The logic stays the same, the tests are not impacted.

Instead of the JVM, we new use native executable thanks to [Quarkus](https://quarkus.io).

Parts of the project that got updated:
- Dockerfile : now build a complete image running the project natively
- Topology creation: now uses [Quarkus Kafka Streams](https://quarkus.io/guides/kafka-streams)
- README: updated to reflect the use of Quarkus
- Github workflows: update of the Docker build job and add a native build job

### CODE REVIEW

**Review by amimart — CHANGES_REQUESTED** [review]
Thanks for such refactoring, native build will bring some significant lightness.

I won't go in detailed remarks for the moment, but here a general thought, I saw the maven publish tasks was removed, but I'd like to keep the Maven publication. From my point of view the quarkus native build should be seen as an option, but not incompatible with standard jar delivery. Regarding the CI we could have different build / publish jobs for jar / native builds.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
         with:
           arguments: build -x check
 
-  build-docker:
+  build-native:
```
Comment (amimart) [inline .github/workflows/build.yml:40]: As we already cover this build type in the docker build, maybe we can avoid it.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -55,10 +55,27 @@ jobs:
...
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-java@v3
         with:
-          arguments: build -x check
+          distribution: 'temurin'
+          java-version: '11'
```
Comment (amimart) [inline .github/workflows/build.yml:71]: I don't think java is needed as we build in docker.

Comment (amimart) [inline .github/workflows/build.yml:71]: I don't think java is needed as we build in docker.

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+# Stage 1 : build
+FROM quay.io/quarkus/ubi-quarkus-native-s2i:22.1-java11 AS build
 
-COPY build/libs/kafka-processor-cosmos-block-*-standalone.jar /opt/kafka-processor-cosmos-block.jar
+COPY --chown=quarkus:quarkus . /usr/src/app
 
-ENTRYPOINT ["java","-jar","/opt/kafka-processor-cosmos-block.jar"]
+WORKDIR /usr/src/app
+
+RUN ./gradlew build -Dquarkus.package.type=native --no-daemon
```
Comment (amimart) [inline Dockerfile:8]: We can bypass the test task here, it is already covered by the Test workflow.

```suggestion
RUN ./gradlew build -x check -Dquarkus.package.type=native --no-daemon
```

Comment (amimart) [inline Dockerfile:8]: We can bypass the test task here, it is already covered by the Test workflow.

```suggestion
RUN ./gradlew build -x check -Dquarkus.package.type=native --no-daemon
```

File: build.gradle.kts [hunk 4]
```diff
@@ -43,6 +38,7 @@ afterEvaluate {
 
 repositories {
     mavenCentral()
+    mavenLocal()
```
Comment (amimart) [inline build.gradle.kts:41]: We do not need the local maven repository.

```suggestion
```

Comment (amimart) [inline build.gradle.kts:41]: We do not need the local maven repository.

```suggestion
```

File: settings.gradle.kts [hunk 5]
```diff
@@ -1 +1,14 @@
+pluginManagement {
+    val quarkusPluginVersion: String by settings
+    val quarkusPluginId: String by settings
+    repositories {
+        mavenLocal()
```
Comment (amimart) [inline settings.gradle.kts:5]: We do not need the local maven repository.

Comment (amimart) [inline settings.gradle.kts:5]: We do not need the local maven repository.

**Review by amimart — CHANGES_REQUESTED** [review]
Great! I just have some minor remarks, opened to discussion 😉

**Review by amimart — DISMISSED** [review]
Looks good to me! Thank you 😉

File: src/test/kotlin/com/okp4/processor/cosmos/TopologyTest.kt [hunk 6]
```diff
@@ -45,8 +45,12 @@ class TopologyTest : BehaviorSpec({
     ).toProperties()
 
     given("A topology") {
-        val topology = topology(config)
-        val testDriver = TopologyTestDriver(topology, config)
+        val topologyProducer = TopologyProducer()
+        topologyProducer.topicIn = "in"
+        topologyProducer.topicOut = "out"
+        topologyProducer.topicError = "error"
```
Comment (ccamel) [inline src/test/kotlin/com/okp4/processor/cosmos/TopologyTest.kt:51]: Just to have something to say! 😁

```suggestion
        val topologyProducer = TopologyProducer().apply {
            topicIn = "in"
            topicOut = "out"
            topicError = "error"
        }
```

**Review by amimart — APPROVED** [review]

**Review by ccamel — APPROVED** [review]
Go! 🚀

**Pull request merged**